File: azor-the-chatdog-py/src/llm/openrouter_validation.py
from pydantic import BaseModel, Field, validator
from typing import Literal


class OpenRouterConfig(BaseModel):
    engine: Literal["OPENROUTER"] = Field(default="OPENROUTER")
    model_name: str = Field(..., description="OpenRouter model name")
    openrouter_api_key: str = Field(..., min_length=1, description="OpenRouter API key")

    @validator('openrouter_api_key')
    def validate_api_key(cls, v):
        if not v or v.strip() == "":
            raise ValueError("OPENROUTER_API_KEY cannot be empty")
        return v.strip()

File: azor-the-chatdog-js/src/llm/openrouterValidation.ts
/**
 * OpenRouter configuration validation using Zod
 */

import { z } from 'zod';

/**
 * OpenRouter configuration schema
 */
export const OpenRouterConfigSchema = z.object({
  engine: z.literal('OPENROUTER').default('OPENROUTER'),
  modelName: z.string().min(1, 'Model name is required'),
  openrouterApiKey: z.string().min(1, 'OpenRouter API key is required'),
});

export type OpenRouterConfig = z.infer<typeof OpenRouterConfigSchema>;

/**
 * Validate and parse OpenRouter configuration from environment
 */
export function validateOpenRouterConfig(): OpenRouterConfig {
  const config = {
    engine: 'OPENROUTER' as const,
    modelName: process.env.OPENROUTER_MODEL_NAME || process.env.MODEL_NAME || 'qwen/qwen3.5-9b',
    openrouterApiKey: process.env.OPENROUTER_API_KEY || '',
  };

  return OpenRouterConfigSchema.parse(config);
}
